src/chart/tree/TreeView.ts

```diff
@@ -603,28 +603,46 @@ function removeNodeEdge(
     data: SeriesData,
     group: graphic.Group,
     seriesModel: TreeSeriesModel,
-    removeAnimationOpt: AnimationOption
+    removeAnimationOpt: AnimationOption,
+    currentSymbol?: TreeSymbol,
+    sourceSymbol?: TreeSymbol
 ) {
     const virtualRoot = data.tree.root;
     const { source, sourceLayout } = getSourceNode(virtualRoot, node);
 
-    const symbolEl: TreeSymbol = data.getItemGraphicEl(node.dataIndex) as TreeSymbol;
+    // use the current symbol when the node is delete immediately
+    const symbolEl: TreeSymbol = data.getItemGraphicEl(node.dataIndex) as TreeSymbol ?? currentSymbol;
 
     if (!symbolEl) {
         return;
     }
 
-    const sourceSymbolEl = data.getItemGraphicEl(source.dataIndex) as TreeSymbol;
-    const sourceEdge = sourceSymbolEl.__edge;
+    // use the source symbol when the node is delete immediately
+    const sourceSymbolEl = data.getItemGraphicEl(source.dataIndex) as TreeSymbol ?? sourceSymbol;
+    const sourceEdge = sourceSymbolEl?.__edge;
+
+
+    const symbolChildren = source.children.map(
+        item => data.getItemGraphicEl(item.dataIndex)
+    ).filter(item => !!item);
+    const notLeaveChildren = symbolChildren.filter(
+        item => !item?.animators?.length || item?.animators?.find(animator => animator.scope !== 'leave')
```

Comment:
It's probaly not a good idea to check `item.animators` to know if there is series animation. You should probably use `animatableModel && animatableModel.isAnimationEnabled()` instead.

Reply:
fixed, use animationModel to check animation state